src/esp/sensor/CameraSensor.cpp

```diff
@@ -150,6 +140,11 @@ bool CameraSensor::drawObservation(sim::Simulator& sim) {
   if (sim.isFrustumCullingEnabled())
     flags |= gfx::RenderCamera::Flag::FrustumCulling;
 
+  // Set projection matrix, transformation matrix, and viewport
+  renderCamera_->setProjectionMatrix(width_, height_, projectionMatrix_);
+  renderCamera_->node().setTransformation(this->computeTransformationMatrix());
```

Comment:
You do not need to set it every time since the renderCamera and the sensor are attaching to the same node so it always has the same transformation as the sensor!